src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -482,98 +461,92 @@ public override async Task RunAsync()
             }
-            else if (isInProc6)
+
+            var isInproc6Runtime = string.Equals(HostRuntime, DotnetConstants.InProc6HostRuntime, StringComparison.OrdinalIgnoreCase);
+            var isInproc8Runtime = string.Equals(HostRuntime, DotnetConstants.InProc8HostRuntime, StringComparison.OrdinalIgnoreCase);
+
+            switch (currentWorkerRuntime)
             {
-                if (await IsInProcNet8Enabled())
-                {
-                    throw new CliException($"The runtime host value passed in, {DotnetConstants.InProc6HostRuntime}, is not a valid host version for your project. For the {DotnetConstants.InProc6HostRuntime} runtime, the {Constants.FunctionsInProcNet8Enabled} variable must not be set while running a .NET 6 in-proc app.");
-                }
-                PrintVerboseForHostSelection(isVerbose, DotnetConstants.InProc6HostRuntime);
-                return;
-            }
-            // Throw an exception if HostRuntime is set to none of the expected values
-            throw new CliException($"Invalid host runtime '{HostRuntime}'. Valid values are 'default', '{DotnetConstants.InProc8HostRuntime}', '{DotnetConstants.InProc6HostRuntime}'.");
+                case WorkerRuntime.dotnet:
+                    if (string.Equals(HostRuntime, "default", StringComparison.OrdinalIgnoreCase))
+                    {
+                        ThrowCliException($"The provided value is only valid for the worker runtime '{WorkerRuntime.dotnetIsolated}'");
+                    }
+
+                    if (isInproc8Runtime && !await validateDotNet8ProjectEnablement())
+                    {
+                        ThrowCliException($"For the {DotnetConstants.InProc8HostRuntime} runtime, the '{Constants.FunctionsInProcNet8Enabled}' environment variable must be set. See https://aka.ms/azure-functions/dotnet/net8-in-process.");
+                    }
+                    else if (isInproc6Runtime && await validateDotNet8ProjectEnablement())
+                    {
+                        ThrowCliException($"For the {DotnetConstants.InProc6HostRuntime} runtime, the '{Constants.FunctionsInProcNet8Enabled}' environment variable cannot be be set. See https://aka.ms/azure-functions/dotnet/net8-in-process.");
+                    }
+                    break;
+                case WorkerRuntime.dotnetIsolated:
+                    if (isInproc8Runtime || isInproc6Runtime)
+                    {
+                        ThrowCliException($"The provided value is only valid for the worker runtime '{WorkerRuntime.dotnet}'");
+                    }
+                    break;
+                default:
+                    ThrowCliException($"The runtime argument is only valid for runtimes '{WorkerRuntime.dotnetIsolated}' and '{WorkerRuntime.dotnet}'");
+                    break;
```

Comment:
This may just be in the logic we're refactoring here, but what is the valid case for specifying the runtime argument for the isolated worker model? Should we just limit this flag to in-proc fully?

Reply:
This logic was present in the initial implementation to support an explicit `default` value. The primary goal of these changes is to refactor the code for better readability and maintainability, while also adding test coverage, preserving the original behavior.

The use case this covers was not defined in previous designs but covers the somewhat unlikely scenario where inference would incorrectly map to an in-proc runtime, enabling an explicit override. Such a case would be a bug, but again, re-evaluating that was not the focus of the current changes, and having the option to avoid inference doesn't case any problems here.

Reply:
I actually ended up collapsing the handling for isolated and non .NET worker cases, simplifying the flow and expanding the runtime support to other stacks (that's something I had explicitly blocked with the changes, differing from the original). This let me clean things up even further.